### PR TITLE
fix(uri-parser): always treat appname as a string

### PR DESCRIPTION
**Pull request by daprahamian**
Fixes NODE-1983

Needs to be ported to `next`

### CODE REVIEW

File: lib/uri_parser.js [hunk 1]
```diff
@@ -124,6 +124,10 @@ function parseQueryStringItemValue(key, value) {
     // deduplicate and simplify arrays
     value = value.filter((v, idx) => value.indexOf(v) === idx);
     if (value.length === 1) value = value[0];
+  } else if (STRING_OPTIONS.has(key)) {
+    // TODO: refactor function to make this early return not
+    // stand out
+    return value;
   } else if (value.indexOf(':') > 0) {
```
Comment (mbroadst) [inline lib/uri_parser.js]: I think we also only want to take this path if the key is `readpreferencetags `

Comment (daprahamian) [reply lib/uri_parser.js]: also `authMechanismProperties`

Comment (mbroadst) [reply lib/uri_parser.js]: should we special case those two explicitly?

Comment (daprahamian) [reply lib/uri_parser.js]: I tried doing it pretty quickly, and it led to errors. I think we should instead make a larger ticket to deal with this, where we hard-code everything from https://github.com/mongodb/specifications/blob/master/source/uri-options/uri-options.rst, and specifically parse based off of that. For now, I'd like to just get this in to fix the ticket.

Comment (mbroadst) [reply lib/uri_parser.js]: OK SGTM

**Pull request merged**